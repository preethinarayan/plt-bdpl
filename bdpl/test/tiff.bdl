struct image_file_header
{
	byte[2] endianness;
	byte[2] magic satisfies {magic[0]==42} else { print("bad magic::");};
	int ifd_pointer;
}header satisfies {1} then {$$=header.ifd_pointer*8;};

struct image_file_directory
{
  byte[2] numEntries;
  struct image_file_directory_entry
  {
	  byte[2] tag;
	  byte[2] typefield;
	  byte[4] count;
	  byte[4] value;
  }[numEntries[0]+numEntries[1]*8] ifdentry satisfies{1} then {print(numEntries[0]+numEntries[1]*8);print("::");};
  int nextifd satisfies { nextifd>0 } then { print(nextifd); $$=nextifd*8;print("::");} ;
}ifd;

file "x:/test/test.tif" inp_file;
read(inp_file,header);
read(inp_file,ifd);
print(ifd);
for(;ifd.nextifd > 0 ;)
{
	read(inp_file,ifd);
	print(ifd);
}


