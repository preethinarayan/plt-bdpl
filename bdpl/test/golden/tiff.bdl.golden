17
struct:image_file_directory
{
	numEntries => ARRAY of TYPE_BYTE
	[
		0 => 0x11,
		1 => 0x0,
	]
	,
	ifdentry => ARRAY of struct:image_file_directory_entry
	[
		0 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x0,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x3,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0xc0,
				1 => 0x6,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		1 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x3,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x48,
				1 => 0x9,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		2 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x2,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x3,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		3 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x3,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x3,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x4,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		4 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x6,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x3,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x0,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		5 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0xa,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x3,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x2,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		6 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0xd,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x2,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0xf,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x16,
				1 => 0x2b,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		7 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0xe,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x2,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x13,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x26,
				1 => 0x2b,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		8 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x11,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x4,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x8,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		9 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x12,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x3,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		10 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x15,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x3,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		11 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x16,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x4,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x48,
				1 => 0x9,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		12 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x17,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x4,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x3c,
				1 => 0x2a,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		13 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x1a,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x5,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x3a,
				1 => 0x2b,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		14 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x1b,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x5,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x42,
				1 => 0x2b,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		15 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x1c,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x3,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
		16 => struct:image_file_directory_entry
		{
			tag => ARRAY of TYPE_BYTE
			[
				0 => 0x28,
				1 => 0x1,
			]
			,
			typefield => ARRAY of TYPE_BYTE
			[
				0 => 0x3,
				1 => 0x0,
			]
			,
			count => ARRAY of TYPE_BYTE
			[
				0 => 0x1,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
			value => ARRAY of TYPE_BYTE
			[
				0 => 0x2,
				1 => 0x0,
				2 => 0x0,
				3 => 0x0,
			]
			,
		}
		,
	]
	,
	nextifd => 0,
}
